core/src/com/po/fuck/model/weapons/Gun.java

```diff
@@ -1,13 +1,26 @@
 package com.po.fuck.model.weapons;
 
 import com.badlogic.gdx.math.Vector2;
+import com.po.fuck.model.sprites.WeaponSpriteInfo;
 import com.po.fuck.model.Entity;
 import com.po.fuck.model.Updatable;
 import com.po.fuck.model.lifetime.Manager;
+import com.po.fuck.model.GeometryMisc;
+
+import static com.po.fuck.assetsManagement.SpriteLoaders.weaponSpriteLoader;
 
 public abstract class Gun extends HandedWeapon implements Updatable {
     protected float cooldown;
     private float cooldownLeft = 0;
+    protected Vector2 muzzlePosition;
+
+    protected Vector2 handedPosition;
+    protected<T>  void setPositions(Class<T> clz) {
+        WeaponSpriteInfo info = weaponSpriteLoader.getSpriteInfo(this.getClass());
+        muzzlePosition = info.muzzlePosition;
+        handedPosition = info.muzzlePosition;
+        geometryData.setSize(info.getSize());
+    }
```

Comment:
I think we can pass `info` here ? And also rename it, because it is setting not only the positions, `geometryData.setSize(info.getSize());` for example.